src/main/java/net/mcreator/ui/blockly/BlocklyJavascriptBridge.java

```diff
@@ -239,6 +241,26 @@ public void addExternalTrigger(ExternalTrigger external_trigger) {
 		ext_triggers.put(external_trigger.getID(), external_trigger.getName());
 	}
 
+	@SuppressWarnings("unused") public String getDependencyType(String procedureName, String dependencyName) {

```

Comment:
Don't you think it would be cleaner to only have one method e.g. `getDependencyMap(String procedureName)` that would return Map<String, Dependency> where the key is the dependency name and the value is the dependency object?

You can then use this method instead of both current methods, and can also make the tooltip contain both dependency names as well as their types.

Reply:
Tool tip already uses `Dependency#toString()` which returns what you suggest.
Also, this is going to be the first method to return something except `boolean`, `String` or an array of those. Can a map like that (or, better, `Map<String, String>`) be directly converted to native JS map object?

Reply:
I do think so.

> Tool tip already uses Dependency#toString() which returns what you suggest.

I see, but it may be better to use a single method for both of these. Passing dependency directly should also be easier in the future if we need to extend something with some additional dependency data that would otherwise require yet another new method

Reply:
I tried to use the map for at least tool tip and there seems to be no way to get values from map in Blockly scripts, neither directly nor via keys, neither using `Object.values(map)` nor `Array.from(map.values())`.

Reply:
![image](https://github.com/MCreator/MCreator/assets/16374228/54e0545f-5b8e-4b3e-957c-28469582573d)

![image](https://github.com/MCreator/MCreator/assets/16374228/64af7f19-3b7a-4f11-8093-579c778fae0b)

![image](https://github.com/MCreator/MCreator/assets/16374228/f3cf39c9-ea5f-43bd-905d-7e8f8c669fb9)

I do believe this work

Reply:
According to my tests, it indeed works with immutable maps like `Map.of()`, but as soon as the same is done using regular `HashMap<>` (**Replace with mutable collection** context action in IDEA), the exact same code logs that `values.toArray()` returns `undefined`. Using `Map.ofEntries()` on an array of map entries causes access errors between Java and JS before we can even get to fetch the list of map values.

Reply:
A simple call to .copyOf seems to do the trick:

![image](https://github.com/MCreator/MCreator/assets/16374228/853146e5-a2c0-42a9-a6eb-1d4407e3ebc2)

Reply:
Yay, we found a solution :D